### ISSUE TITLE
WarGames music & resolution

### ISSUE THREAD

**Issue by Scarthcaroth** [elishacloud/dxwrapper#119]
Hi there, fantastic fix for WarGames, I finally have it running of Windows 10!! However the no-CD fix doesn't work. That's a minor issue, but even when I play the game (using mounted ISO) there's no music.

Also I was wondering if it's possible to force the game to a higher resolution?

Cheers :)

**Comment by mirh** [comment]
Yes it's [possible](https://www.vogons.org/viewtopic.php?f=59&t=34931&start=2920#p565759), though idk with dxwrapper. 
Also, missing audio from the iso sounds like having ripped CD-A tracks badly (which I [guess](http://forum.arena80.it/index.php?topic=8625.0#msg72828) the game was using)

**Comment by Scarthcaroth** [comment]
Ty for the reply. Putting graphics/resolution aside, and the fact the no-CD patch also doesn't work for me: I am just trying to get the music to work here.

I tried with my real CD and the music plays fine. It is using CD-A/Redbook, because when I try a normal music CD it will play those tracks too.

But no matter how I seem to rip the CD image, trying bin/cue files and other formats, using PowerISO and IMGBurn and DaemonTools, I just can't get the game to play audio when mounting the image.

I suspect I can fix this by editing the CUE file somehow, but I'm clueless to do that. I did manage to extract the game's music tracks (into WAV format) and from what I understand, I can point the CUE file to those files, I just don't know how.

It would be great to get this game fully working again, since we'll never see it on GoG (MGM Interactive apparently hate videogames)

**Comment by Scarthcaroth** [comment]
I should add, the contents on my CUE file:

FILE "WarGames.bin" BINARY
  TRACK 01 MODE1/2352
    INDEX 01 00:00:00
  TRACK 02 AUDIO
    PREGAP 00:02:00
    INDEX 01 28:44:06
  TRACK 03 AUDIO
    INDEX 00 31:49:43
    INDEX 01 31:51:43
  TRACK 04 AUDIO
    INDEX 00 35:55:65
    INDEX 01 35:57:65
  TRACK 05 AUDIO
    INDEX 00 40:01:27
    INDEX 01 40:03:27
  TRACK 06 AUDIO
    INDEX 00 44:05:69
    INDEX 01 44:07:69
  TRACK 07 AUDIO
    INDEX 00 48:19:27
    INDEX 01 48:21:27
  TRACK 08 AUDIO
    INDEX 00 52:23:34
    INDEX 01 52:25:34
  TRACK 09 AUDIO
    INDEX 00 56:28:49
    INDEX 01 56:30:49
  TRACK 10 AUDIO
    INDEX 00 60:37:15
    INDEX 01 60:39:15

**Comment by mirh** [comment]
https://github.com/bangstk/ogg-winmm
https://community.pcgamingwiki.com/files/file/107-patched-_inmmdll/
.. or maybe try your ISO mounting program? Can you open the tracks with a normal audio player?

**Comment by Scarthcaroth** [comment]
I can't load the bin/cue files into my media player to play as audio CD tracks. As soon as I mount them it seems to detect the image as a mixed audio/data type at first, but 1 second later it just sees it as a data disc. I have no idea why. I just can't seem to get a normal working image file from my physical disc no matter what I try.

I will try this winmm thing, I just hope that by doing that the game won't ask me to insert the CD to play it...

**Comment by mirh** [comment]
Sounds like you aren't ripping the Mixed Mode CD properly. 
Btw, those dll hacks assume that there isn't other data to be read from the original cd, other than music.

**Comment by Scarthcaroth** [comment]
I am really trying, all this stuff is really complicated. Not sure about this winmm thing, just getting to the DLL file download was really hard for me >_< I am trying to install it now to play the off files I managed to extract from the CD image

**Comment by Scarthcaroth** [comment]
By the way, what is the _inmm.dll program for? I installed it but no idea what it's for. And I tried following the instructions for winmm.dll to play ogg files for the game exactly, it doesn't work though :/

**Comment by Scarthcaroth** [comment]
I have reached the point where not even my official, retail CD is recognised anymore as containing playable audio files in media player, or having an autoplay setup. It's just generic storage now :/ But it still works to play the game and the music works in-game too. I guess I just need help in how to properly rip the image from it. I've tried many programs, ISOBuster being the most recent, to get bin and cue files - still no luck

**Comment by Scarthcaroth** [comment]
Nevermind, I've basically given up. Fixed my CD drive not recognising CDs by reinstalling the CD-ROM driver. Uninstalled all the above stuff. I can still play the game with music only by using my physical CD, or I can play the game using an image and for music I have to just play the wav/ogg files in the background. Even using the real CD, the tracks never advance or loop anyway.

I just wish there was better treatment for this game. It's got hardly any love. I'd love to see it with better graphics (just higher res) and proper working music and no-CD capability. Cheers for the help

**Comment by mirh** [comment]
Mixed [is a mess](http://wiki.redump.org/index.php?title=CD_Dumping_Guide_with_Audio_Tracks_\(Old\)).. But did you try CD BurnerXP, ImgBurn or DiscImageCreator?
Also see [this](https://www.pcgamingwiki.com/wiki/Troubleshooting_guide/Audio_problems#CD_music_not_playing), I have never heard of drive drivers.

**Comment by elishacloud** [comment]
The CD fix is specific to the exact exe file used for the game.  Can you upload you exe file and I will check it out?

**Comment by Scarthcaroth** [comment]
Totally forgot about this thread! I am still having the same issue so I can send my exe file a bit later today :)

**Comment by Scarthcaroth** [comment]
[wargames.zip](https://github.com/user-attachments/files/18002328/wargames.zip)
Hi again, here is my exe file. Thank you!

**Comment by elishacloud** [comment]
Ok, this patch should fix the problem:  [dxwrapper.zip](https://github.com/user-attachments/files/18017594/dxwrapper.zip)

Note: make sure you follow all the steps [here](https://github.com/elishacloud/dxwrapper/wiki/WarGames) for the no-CD patch.

**Comment by Scarthcaroth** [comment]
Hmm that didn't work, the game just doesn't boot at all after selecting Software Rasteriser. Whether or not I have the ISO mounted to my virtual CD drive.

If I revert back to the old 3 dx files you previously provided which I've had for ages, it does show the EA logo, then as soon as the MGM logo appears it quits. I tried this with and without the movie files from the CD copied, and in both cases the game doesn't get past the EA logo.

So I can't play the game with or without the CD :( I'm just running Windows 10. It worked before, years ago, maybe needs reinstalling but I doubt it

**Comment by elishacloud** [comment]
I tested with your executable so I know it works.  I suspect that there is something wrong with the rest of the files in the game folder.  Make sure you do the following:

1. Install WarGames to a folder on your computer.  Run the WarGames installer to do a full install into that folder.
2. Copy 'briefres', 'data' and 'video' folders from CD to the game folder.
3. Unzip the dxwrapper files I gave you yesterday to the game folder.

The folder should look something like this when you are done:

![image](https://github.com/user-attachments/assets/fed1489c-e453-4689-b40a-19abc7887f29)

**Comment by Scarthcaroth** [comment]
Ok I have done all that, launched the game and after choosing Primary display diver and clicking OK, I get a black screen for 1 second then it just quits back to Windows. This is with the CD ISO mounted or no drive mounted.  I tried running as admin, and compatibility settings (Win 98).

During installation I opted not to install Direct3D, thinking there surely would already be D3D drivers on modern Windows.

At this point I can't play the game at all, and I'd like to even if I had to mount the CD.

**Comment by elishacloud** [comment]
> At this point I can't play the game at all, and I'd like to even if I had to mount the CD.

If you are trying to play with the CD then you will need to disable the no-CD hack.  Just add semi-colon in front of the `VerificationAddress`.

> I can give you my log file if that helps, it contains the errors:

Yes, please upload the full log file.  That may help.

Does your monitor take a couple seconds to switch resolutions?  I have seen issues with older games when monitors that do that.

**Comment by elishacloud** [comment]
You can try this one and see if it helps.  I added some code to check for the kind of device creation failure you are getting:  [dxwrapper.zip](https://github.com/user-attachments/files/18046444/dxwrapper.zip)

If this doesn't work please attach your full log file.

**Comment by Scarthcaroth** [comment]
Hi again, 
Those last files also didn't work, here's my log file. The screen adjusts resolution, then just goes black after clicking OK on Software Rasteriser, and then quits back to Windows.
[dxwrapper-wargames.log](https://github.com/user-attachments/files/18221542/dxwrapper-wargames.log)

**Comment by StefanStanic95** [comment]
For me, the only thing that worked to launch the game on Windows 10 is the version of the patch from 2019, available here:
https://www.gamepressure.com/download/wargames-defcon-1-windows-10-fix-mod/z5128a9

The latest one from https://github.com/elishacloud/dxwrapper/wiki/WarGames changes the resolution to 640x480 and crashes to desktop immediately after selecting Software Rasteriser.

EDIT 1:
It seems that the [HotPatch] section of the dxwrapper.ini file causes the crash for me. After removing it or replacing it with the section from the 2019 version of dxwrapper.ini, the game no longer crashes on startup using the latest wrapper. I've tested on both Windows 10 and Windows 11 successfully.

EDIT 2:
My game does not crash when Windows 98 and 16-bit color compatibility is enabled in properties.

**Comment by elishacloud** [comment]
@StefanStanic95, Did you follow [the steps](https://github.com/elishacloud/dxwrapper/wiki/WarGames#installation) by copying all the files from your Wargames CD onto the Wargames folder?  That hotpatch code will bypass the CD code and just look for the files on your hard disk instead.

**Comment by StefanStanic95** [comment]
> [@StefanStanic95](https://github.com/StefanStanic95), Did you follow [the steps](https://github.com/elishacloud/dxwrapper/wiki/WarGames#installation) by copying all the files from your Wargames CD onto the Wargames folder? That hotpatch code will bypass the CD code and just look for the files on your hard disk instead.

Yes, I did. I downloaded the latest WarGames-fix.zip and noticed that the lines under [HotPatch] are now commented out. I uncommented them all out, and the game would not start after that. Perhaps my game version is wrong.

My wargames.exe has the following characteristics:
Modification date: ‎Monday, ‎July ‎20, ‎1998, ‏‎10:49:42 AM
Version 1.0.0.1
Product version 970430.1
MD5 hash 3952ddd6115993da8411ea237a5bd498

**Comment by elishacloud** [comment]
> Yes, I did. I downloaded the latest WarGames-fix.zip and noticed that the lines under [HotPatch] are now commented out. I uncommented them all out, and the game would not start after that. Perhaps my game version is wrong.

Try commenting out the lines under [HotPatch].  They cause issues with some versions of the game.

**Comment by Scarthcaroth** [comment]
Just came here to say that just like @StefanStanic95 I have managed to get WarGames playing on Windows 11, but with the caveats that it's stuck in 640x640 resolution, and the CD skip hotfix doesn't work. I can play it using a mounted ISO image I created with IMGBurn, but music won't work. With my retail CD, the music does work.

So this seems to be the best for now. But I'd really love to increase that resolution. No matter what I try in the ini file, it just sticks to 640x640 every time.

**Comment by elishacloud** [comment]
> but with the caveats that it's stuck in 640x640 resolution

As far as I know this game only supports 640x480 resolution.  So this is normal.

> and the CD skip hotfix doesn't work.

Yeah.  There are several versions of the game.  I tested 3 versions but the patch may only work on some of the versions of WarGames.

**Comment by Linaromew** [comment]
It looks like on AMD cards (RX7900XTX), the provided dxwrapper config on the Wargames page doesn't quite work: it now requires **DdrawIntroVideoFix** and **DdrawEnableByteAlignment**. Without them the game crashes in the intro video, and after that was drawing a completely black screen with sound. Odd one but hopefully saves the headache for anyone else running into it; I didn't need to do the same for an Intel machine.

On an unrelated note, since nobody else has even heard of this game and there's no communities around it, there was a patch which added some fairly substantial stuff:

https://web.archive.org/web/20000302231519/http://www.wargamesmgm.com/

But it doesn't look like it was *ever* archived anywhere; if anyone ever manages to locate it that'd be pretty awesome...

**Comment by Scarthcaroth** [comment]
> > but with the caveats that it's stuck in 640x640 resolution
> 
> As far as I know this game only supports 640x480 resolution. So this is normal.
> 
> > and the CD skip hotfix doesn't work.
> 
> Yeah. There are several versions of the game. I tested 3 versions but the patch may only work on some of the versions of WarGames.

Sorry, yes I meant I am getting 640x480.

**Comment by Scarthcaroth** [comment]
> It looks like on AMD cards (RX7900XTX), the provided dxwrapper config on the Wargames page doesn't quite work: it now requires **DdrawIntroVideoFix** and **DdrawEnableByteAlignment**. Without them the game crashes in the intro video, and after that was drawing a completely black screen with sound. Odd one but hopefully saves the headache for anyone else running into it; I didn't need to do the same for an Intel machine.
> 
> On an unrelated note, since nobody else has even heard of this game and there's no communities around it, there was a patch which added some fairly substantial stuff:
> 
> https://web.archive.org/web/20000302231519/http://www.wargamesmgm.com/
> 
> But it doesn't look like it was _ever_ archived anywhere; if anyone ever manages to locate it that'd be pretty awesome...

This is really interesting! I somewhat remember something about a patch back in the day. I searched my PC but I sadly don't have it. The file in question is called wgpi740.zip. Searching Google and another internet archive yielded no results. I scoured that website and found the name of the Producer at MGM Interactive at the time, Robb Alvey. He has a YouTube channel and website and facebook, all about reviewing Theme Parks. I contacted him on his website and his Facebook: https://www.facebook.com/themeparkreview and asked if he has the file or knows someone who might. Here's hoping for a response!

At this point, the people in this thread are pretty much the whole WarGames community, but I did also find a small thread here https://www.myabandonware.com/game/wargames-evk this thread has modern compatibility patches, but none are the official patch sadly.

**Comment by Linaromew** [comment]
![Image](https://github.com/user-attachments/assets/00605668-c2a2-46c7-9b2c-03576e4fed4a)

On your resolution note - some time ago I was working on adding higher resolution support, hacking in the UI positions and fixing some bugs etc, but got stopped by the AMD issue after an upgrade, so hoping to pick it up again now that things are playing nice.

But yeah, as Elishacloud mentioned, the entire game is _completely hardcoded_ to 640x480, down to UI positions and how intros work, so it's a bit of a mess :)

**Comment by elishacloud** [comment]
> It looks like on AMD cards (RX7900XTX), the provided dxwrapper config on the Wargames page doesn't quite work: it now requires **DdrawIntroVideoFix** and **DdrawEnableByteAlignment**. Without them the game crashes in the intro video, and after that was drawing a completely black screen with sound.

Thanks for testing this.  I updated the [wiki](https://github.com/elishacloud/dxwrapper/wiki/WarGames) with new patch that has these options enabled by default.

_Note: `DdrawEnableByteAlignment` used to be hard coded enabled but it caused issues with some games and I was able to fix most of the games that required it.  `DdrawIntroVideoFix` may not have been needed in older versions of dxwrapper because of the way it worked.  However, a few fixes made me change the way that I had to handle these cases._

**Comment by elishacloud** [comment]
Closing issue.  The latest wiki should have all the fixes for this game.  Please reopen if you still have an issue.
